Source/WebCore/Modules/applepay/ApplePaySession.cpp

```diff
@@ -332,6 +349,15 @@ static ExceptionOr<ApplePaySessionPaymentRequest> convertAndValidate(Document& d
     }
 #endif
 
+#if ENABLE(APPLE_PAY_DISBURSEMENTS)
+    if (paymentRequest.disbursementPaymentRequest) {
+        auto disbursementPaymentRequest = WTFMove(*paymentRequest.disbursementPaymentRequest).convertAndValidate();
```

Comment:
Doesn't the && qualifier for convertAndValidate do the move for us here? So the `WTFMove(*paymentRequest.disbursementPaymentRequest)` is redundant?

Reply:
Actually the more I think about this the less I like it. Why are we moving an empty struct? Moving it isn't doing anything. This would make more sense if the struct had some well-defined but invalid state that a moved-from object would be in. But, this is an empty struct which has no validation logic.

All that is to say all this moving and `foo() &&` stuff is complicated for no real benefit. I suggest either defining an invalid state that a moved-from object is in and check for that in `validate` or remove all this moving alltogether.